### PR TITLE
feat(KNO-4622) Add "knock commit get" command

### PR DESCRIPTION
**Pull request by francoborr**
### Description
Adds a `knock commit get` command for retrieving one single commit based on an ID.

### Tasks
[KNO-4622](https://linear.app/knock/issue/KNO-4622/[cli]-add-knock-commits-get-command)

### CODE REVIEW

File: src/commands/commit/get.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        key: "Identifier",
+        value: commit.resource.identifier,
+      },
+      {
+        key: "Author",
+        value: formatCommitAuthor(commit),
+      },
+      {
+        key: "Commit message",
+        value: commit.commit_message ? commit.commit_message.trim() : "",
```
Comment (thomaswhyyou) [inline src/commands/commit/get.ts:59]: I believe we still have the same issue here though, where a long commit message still gets truncated?

CleanShot 2023-11-13 at 10 05 38@2x

I'm thinking we might want to follow the similar format as git, where we basically print out the commit message separately at the bottom outside a table.

File: test/lib/api-v1.test.ts [hunk 2]
```diff
@@ -226,6 +226,25 @@ describe("lib/api-v1", () => {
     });
   });
 
+  describe("getCommit", () => {
```
Comment (thomaswhyyou) [inline test/lib/api-v1.test.ts:229]: Not on this PR, but following up on my [comment](https://github.com/knocklabs/control/pull/2598/files#r1389696396) from the control side PR: would you mind opening a PR and adding integration tests for the get endpoint if missing?

**Review by thomaswhyyou — COMMENTED** [review]
A couple comments/questions!

File: test/lib/api-v1.test.ts (hunk 2 above)
Comment (francoborr) [reply test/lib/api-v1.test.ts:229]: I think these are the ones you are looking for right? https://github.com/knocklabs/control/blob/main/backend/test/control_web/controllers/v1/commit_controller/show_commit_test.exs

Comment (thomaswhyyou) [reply test/lib/api-v1.test.ts:229]: Dohh, you are right we got it covered. Disregard 😅

File: src/commands/commit/get.ts (hunk 1 above)
Comment (francoborr) [reply src/commands/commit/get.ts]: Here's the commit: https://github.com/knocklabs/knock-cli/pull/263/commits/1dd6ca9ba4e3af0b1471bb29f2b4157d53be18ad

And this is how it looks now:

Screen Shot 2023-11-13 at 14 16 04
Screen Shot 2023-11-13 at 14 15 57

Another option could be to show `Commit Message: <commit.commit_message>`, but I think how we have it now looks better.

File: src/commands/commit/get.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        header: "",
+        minWidth: 16,
+      },
+    });
+
+    this.log("");
+
+    if (commit.commit_message) {
+      ux.info(commit.commit_message);
+      this.log("");
```
[the remark this answers is not in the document]
Comment (francoborr) [reply src/commands/commit/get.ts:80]: This Is for having a space between the commit message and the terminal input, I think it looks nicer (we also have a new line on the top):

Screen Shot 2023-11-13 at 15 18 13

[the remark this answers is not in the document]
Comment (francoborr) [reply src/commands/commit/get.ts:80]: Is for having some space between the commit message and the terminal input, I think it looks nicer.

**Pull request merged**